designs/2019-move-to-async-api/README.md

```diff
@@ -0,0 +1,469 @@
+- Start Date: 2019-09-28
+- RFC PR: https://github.com/eslint/rfcs/pull/40
+- Authors: Toru Nagashima ([@mysticatea](https://github.com/mysticatea))
+
+# `LinterShell` Class Replacing `CLIEngine`
+
+## Summary
+
+This RFC adds a new class `LinterShell` that provides asynchronous API and deprecates `CLIEngine`.
```

Comment:
I know there has already been discussion around this, but I’m not convinced about the name `LinterShell` because I don’t think it’s very descriptive. If I saw this without any context, I don’t think I would have any idea what it actually does.

If anything, it feels like this class should be called `Linter` and the current `Linter` class should be called `LinterEngine`, though that doesn’t seem possible since it’s a big breaking change.

I don’t want to block this RFC for this, but I’d love for us to keep thinking about alternatives.

Reply:
My proposal was:

1. The new class is named `ESLint` as the facade class. (rejected)
2. `CLIEngine` => `LinterShell` and `Linter` => `LinterKernel`, as inspired by shell and kernel. (partly accepted)

Reply:
I actually like `ESLint` as well. I'm happy to go with consensus on this.

Reply:
Thank you.

I have added a new section about the class name (https://github.com/eslint/rfcs/pull/40/commits/ef4d198898ef41e1c4859cd09dd876e8a9192075).

@eslint/eslint-tsc May we go in this direction?